src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingScriptOperation.cs

```diff
@@ -218,16 +218,40 @@ private SqlScriptPublishModel BuildPublishModel()
                     "Scripting object count {0}, objects: {1}",
                     selectedObjects.Count(),
                     string.Join(", ", selectedObjects)));
-
+            
+            string server = GetServerNameFromLiveInstance(this.Parameters.ConnectionString);
             string database = new SqlConnectionStringBuilder(this.Parameters.ConnectionString).InitialCatalog;
+
+
             foreach (ScriptingObject scriptingObject in selectedObjects)
             {
-                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(database));
+                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(server, database));
             }
 
             return publishModel;
         }
 
+        private string GetServerNameFromLiveInstance(string connectionString)
+        {
+            string serverName = string.Empty;
```

Comment:
>string serverName = string.Empty; [](start = 12, length = 33)

nit: you can declare and assign this variable in one statement....removes one line.